jira_bot.py

```diff
@@ -0,0 +1,160 @@
+    except Exception as e:
+        print(f"Error loading YAML file '{file_path}': {e}", file=sys.stderr)
+        return {}
+
+
+def get_jira_username(github_nick, mapping):
+    """
+    Find the Jira username corresponding to a GitHub nickname.
+    """
+    return mapping.get(github_nick, None)
+
+
+def is_epic_issue(jira, issue_key):
+    """
+    Check if a Jira issue exists and is of the type 'Epic'.
+    """
+    try:
+        # Get the issue
+        print(f"Check if issue '{issue_key}' exists.", file=sys.stderr)
+        issue = jira.issue(issue_key)
+
+        # Check if the issue type is 'Epic'
+        print(f"Check if issue '{issue_key}' is an Epic.", file=sys.stderr)
+        if issue.fields.issuetype.name.lower() == 'epic':
+            return True
+        else:
+            return False
+    except Exception as e:
+        print(f"Error: {e}", file=sys.stderr)
+        return False
+
+
+# pylint: disable=too-many-arguments
+def create_jira_task(token, project_key, summary, description, issue_type, epic_link, component, assignee, story_points):
+    """
+    create_jira_task creates a jira issue with the given parameter
+    """
+    options = {
+        'server': JIRA_SERVER,
+        'headers': {
+            'Authorization': f'Bearer {token}'
+        }
+    }
+    try:
+        jira = JIRA(options=options)
+        print("Connected to Jira successfully using a personal access token.", file=sys.stderr)
+    # pylint: disable=broad-exception-caught
+    except Exception as e:
+        print(f"🔴 Failed to connect to Jira: {e}", file=sys.stderr)
+        return
+
+    # Check if Epic exists
+    if is_epic_issue(jira, epic_link) is False:
+        print(f"🔴 The Jira issue '{epic_link}' does not exist or is not of issuetype Epic.", file=sys.stderr)
+        sys.exit(1)
+
+    # Task creation dictionary
+    issue_dict = {
+        'project': {'key': project_key},
+        'summary': summary,
+        'description': description,
+        'issuetype': {'name': issue_type},
+        'customfield_12310243': story_points,
+    }
+
+    # Add epic link if provided
+    if epic_link:
+        issue_dict['customfield_12311140'] = epic_link
+
+    # Add assignee if provided
+    if assignee:
+        issue_dict['assignee'] = {'name': assignee}
+
+    # Add component if provided
+    if component:
+        issue_dict['components'] = [{'name': component}]
+
+    try:
+        new_issue = jira.create_issue(fields=issue_dict)
+        print(f"🟢 Task created successfully: {new_issue.key}", file=sys.stderr)
+        print(new_issue.key)
+    # pylint: disable=broad-exception-caught
+    except Exception as e:
+        print(f"🔴 Failed to create task: {e}", file=sys.stderr)
+
+
+def main():
+    """ main - command line parsing and calling the bot """
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description="Create a Jira task.")
+    parser.add_argument('--token', required=True,
```

Comment:
Should we pass the token in an environment variable instead? That's generally safer than passing it via `argv` (it's at least not visible in `/proc`).

Reply:
How about doing that as a follow-up, because it also affects other scripts in the repo (namely `pr_best_practices.py`)?